examples/medplum-chart-demo/src/bots/core/charting-utils.ts

```diff
@@ -0,0 +1,294 @@
+  };
+  // Create a condition for the encounter diagnosis
+  const encounterDiagnosis: Condition = {
+    resourceType: 'Condition',
+    subject: encounter.subject as Reference<Patient>,
+    encounter: { reference: getReferenceString(encounter) },
+    recorder: { reference: getReferenceString(user) },
+    asserter: { reference: getReferenceString(user) },
+    code: conditionData.reasonForVisit,
+    category: [
+      {
+        coding: [
+          {
+            system: 'http://hl7.org/fhir/ValueSet/condition-category',
+            code: 'encounter-diagnosis',
+            display: 'Encounter Diagnosis',
+          },
+        ],
+      },
+    ],
+  };
+
+  entries.push({ fullUrl: generateId(), request, resource: encounterDiagnosis });
+
+  // If the problem list question was checked, create an additional condition for it
+  if (conditionData.problemList) {
+    entries.push({
+      fullUrl: generateId(),
+      request,
+      resource: {
+        resourceType: 'Condition',
+        subject: encounter.subject as Reference<Patient>,
+        encounter: { reference: getReferenceString(encounter) },
+        recorder: { reference: getReferenceString(user) },
+        asserter: { reference: getReferenceString(user) },
+        code: conditionData.reasonForVisit,
+        category: [
+          {
+            coding: [
+              {
+                system: 'http://hl7.org/fhir/ValueSet/condition-category',
+                code: 'problem-list-item',
+                display: 'Problem List Item',
+              },
+            ],
+          },
+        ],
+      },
+    });
+  }
+
+  return entries;
+}
+
+/**
+ * This function takes ClinicalImpression data and creates a bundle entry so that it can be added to a batch transaction. The
+ * ClinicalImpression resource represents any notes on an encounter in this context.
+ *
+ * @param clinicalImpressionData - Data object containing codes and values for the ClinicalImpression resources
+ * @param encounter - The encounter that the data is derived from
+ * @param user - The user creating the ClinicalImpressions
+ * @returns A bundle entry with the ClinicalImpression resource that can be used in a batch transaction
+ */
+export function createClinicalImpressionEntry(
+  clinicalImpressionData: ClinicalImpressionData,
+  encounter: Encounter,
+  user: Practitioner
+): BundleEntry | undefined {
+  if (!clinicalImpressionData.assessment) {
+    return undefined;
+  }
+
+  // Create the clinical impression
+  const clinicalImpression: ClinicalImpression = {
+    resourceType: 'ClinicalImpression',
+    status: 'in-progress',
+    subject: encounter.subject as Reference<Patient>,
+    encounter: { reference: getReferenceString(encounter) },
+    assessor: { reference: getReferenceString(user) },
+    note: [{ text: clinicalImpressionData.assessment }],
+  };
+
+  // Return the clinical impression in a bundle entry
+  return {
+    fullUrl: generateId(),
```

Comment:
not sure that `generateId()` does the right thing here. Might be better to use `` `urn:uuid:${randomUUID()}` ``